RLBotFramework/utils/class_importer.py

```diff
@@ -1,75 +1,74 @@
 import importlib
 import inspect
-
 import os
+import sys
 
 from RLBotFramework.agents.base_agent import BaseAgent
-from RLBotFramework.utils.logging_utils import log_warn
 
 
-def is_extends_base_class(module, base_class):
-    super_classes = module.__bases__
-    for super_class in super_classes:
-        if super_class == base_class:
-            return True
-    return False
+class AgentLoadData:
+    def __init__(self, python_file, base_class):
+        self.python_file = python_file
+        self.base_class = base_class
+        self.agent_class = None
+        self.agent_module = None
+        self.reload()

```

Comment:
are we sure we want to call reload in the constructor?
It could potentially cause issues.

Also how does this work with the importing of extensions

Reply:
Doesn't seem particularly dangerous to me. Is there something especially likely or especially bad about constructor crashes?

Reply:
I'm pretty sure the object doesn't get created.

Reply:
I think that in the old days, exceptions coming from a constructor were harder to handle properly or harder to debug, so people started giving advice about making constructors do very little. I don't have any reason to believe that such advice is relevant in Python.

Reply:
I don't see there being an actual issue with this, I'm just answering the question. It looks fine.